frontend/packages/console-dynamic-plugin-sdk/docs/api.md

```diff
@@ -1,214 +1,160 @@
](#isallnamespaceskey), [BlueInfoCircleIcon](#blueinfocircleicon), [checkAccess](#checkaccess), [ErrorStatus](#errorstatus), [GenericStatus](#genericstatus), [GreenCheckCircleIcon](#greencheckcircleicon), [InfoStatus](#infostatus), [PopoverStatus](#popoverstatus), [ProgressStatus](#progressstatus), [StatusComponent](#statuscomponent), [StatusIconAndText](#statusiconandtext), [SuccessStatus](#successstatus), [CamelCaseWrap](#camelcasewrap), [ListPageBody](#listpagebody), [CodeEditor](#codeeditor), [consoleFetch](#consolefetch), [consoleFetchJSON](#consolefetchjson), [consoleFetchText](#consolefetchtext), [DocumentTitle](#documenttitle), [ErrorBoundaryFallbackPage](#errorboundaryfallbackpage), [getAPIVersionForModel](#getapiversionformodel), [getGroupVersionKindForModel](#getgroupversionkindformodel), [getGroupVersionKindForResource](#getgroupversionkindforresource), [HorizontalNav](#horizontalnav), [InventoryItem](#inventoryitem), [InventoryItemBody](#inventoryitembody), [InventoryItemLoading](#inventoryitemloading), [InventoryItemStatus](#inventoryitemstatus), [InventoryItemTitle](#inventoryitemtitle), [k8sCreate](#k8screate), [k8sDelete](#k8sdelete), [k8sGet](#k8sget), [k8sList](#k8slist), [k8sListItems](#k8slistitems), [k8sPatch](#k8spatch), [k8sUpdate](#k8supdate), [ListPageCreate](#listpagecreate), [ListPageCreateButton](#listpagecreatebutton), [ListPageCreateDropdown](#listpagecreatedropdown), [ListPageCreateLink](#listpagecreatelink), [ListPageHeader](#listpageheader), [NamespaceBar](#namespacebar), [Overview](#overview), [OverviewGrid](#overviewgrid), [QueryBrowser](#querybrowser), [RedExclamationCircleIcon](#redexclamationcircleicon), [ResourceEventStream](#resourceeventstream), [ResourceIcon](#resourceicon), [ResourceLink](#resourcelink), [ResourceStatus](#resourcestatus), [ResourceYAMLEditor](#resourceyamleditor), [StatusPopupItem](#statuspopupitem), [StatusPopupSection](#statuspopupsection), [TableData](#tabledata), [Timestamp](#timestamp), [useAccessReview](#useaccessreview), [useActiveColumns](#useactivecolumns), [useActiveNamespace](#useactivenamespace), [useFlag](#useflag), [useOverlay](#useoverlay), [usePrometheusPoll](#useprometheuspoll), [useActivePerspective](#useactiveperspective), [useAnnotationsModal](#useannotationsmodal), [useDeleteModal](#usedeletemodal), [useK8sModel](#usek8smodel), [useK8sModels](#usek8smodels), [useK8sWatchResource](#usek8swatchresource), [useK8sWatchResources](#usek8swatchresources), [useLabelsModal](#uselabelsmodal), [useQuickStartContext](#usequickstartcontext), [useResolvedExtensions](#useresolvedextensions), [useUserSettings](#useusersettings), [YellowExclamationTriangleIcon](#yellowexclamationtriangleicon), [PerspectiveContext](#perspectivecontext), [VirtualizedTable](#virtualizedtable), [ListPageFilter](#listpagefilter), [useListPageFilter](#uselistpagefilter), [YAMLEditor](#yamleditor), [useModal](#usemodal), [useAccessReviewAllowed](#useaccessreviewallowed), [useSafetyFirst](#usesafetyfirst) |
```

Comment:
I noticed there's code to sort the APIs below, but these aren't alphabetized.

Also -- maybe for a future update -- it would be nice to somehow catergorize APIs into hooks, components, etc. if there is a way. Maybe something we can put in the jsdoc?

Reply:
@spadgett API symbols are sorted as follows

```ts
apis: getConsolePluginAPIs()
  // Sort APIs by name, list non-deprecated APIs first
  .sort((a, b) => {
    if (a.doc.deprecated !== b.doc.deprecated) {
      return a.doc.deprecated ? 1 : -1;
    }

    return a.name.localeCompare(b.name);
  }),
```

Non-deprecated API symbols are listed before deprecated ones, so the list isn't alphabetized per name only.

Reply:
@spadgett 

> Also -- maybe for a future update -- it would be nice to somehow catergorize APIs into hooks, components, etc. if there is a way. Maybe something we can put in the jsdoc?

Agreed, but that can be done as a follow-up.

In this PR, we categorize all API symbols according to their TS language declaration kind
```ts
declarationKinds: ['Variable', 'TypeAlias', 'Interface', 'Enum'],
```

In future, we can refine this categorization as needed, e.g. add logic to detect React hooks or components etc.

Reply:
@vojtechszocs But if you look at the output, it's not really alphabetical even for non-deprecated APIs